### DISCUSSION TITLE
Help with -fPIC error when building OpenSN

### DISCUSSION THREAD

**Discussion by hityyds** [Open-Sn/opensn#450] [Ask for Help] [answered]
Hello,

I have been following the official installation guide [(https://open-sn.github.io/opensn/install_linux.html)](https://open-sn.github.io/opensn/install_linux.html) for OpenSN and everything goes well until the final `make` command, where I encounter the following error:
```
[ 90%] Linking CXX shared library libopensnlua.so
/usr/bin/ld: /home/zhanghu/ProgramFiles/OpenSN/dependencies/lib/liblua.a(liolib.o): warning: relocation against `stdout@@GLIBC_2.2.5' in read-only section `.text'
/usr/bin/ld: /home/zhanghu/ProgramFiles/OpenSN/dependencies/lib/liblua.a(lstate.o): relocation R_X86_64_PC32 against symbol `lua_newstate' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
make[2]: *** [lua/CMakeFiles/libopensnlua.dir/build.make:1223: lua/libopensnlua.so.0.0.1] Error 1
make[1]: *** [CMakeFiles/Makefile2:191: lua/CMakeFiles/libopensnlua.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```
From the error message, it seems like my Lua installation was not compiled with the `-fPIC` option. However, I am sure that I used the `-fPIC` flag during the installation, and I have also tried recompiling Lua with `-fPIC` multiple times, but the issue persists.

Can anyone help me with this problem? What should I do to resolve it?

Thanks in advance for your assistance!

**Comment by andrsd** [comment, marked as answer]
If you copied the command from step 1.4, I noticed there is a typo:

It should be:
```
make linux MYCFLAGS=-fPIC -j
```

Can you try to re-run step 1.4 with the above. Also, do not forget the second step with `make install`.

**Reply by hityyds** [reply]
Thank you so much for your help! You're absolutely right—I copied the command directly from the tutorial, and there was indeed a typo. After modifying it as you suggested, the issue has been resolved.

**Reply by andrsd** [reply]
Good to hear!
And thanks for reporting back that fixed the problem :+1:
